pyboy/pyboy.py

```diff
@@ -224,12 +227,16 @@ def stop(self, save=True):
             save (bool): Specify whether to save the game upon stopping. It will always be saved in a file next to the
                 provided game-ROM.
         """
-        if not self.stopped:
+        # If __init__ errors, __del__ will call try to call stop() but
+        # self.stopped doesn't exist yet
+        if hasattr(self, "stopped") and not self.stopped:
```

Comment:
Does `hasattr` work in Cython?

Reply:
It seems to? I'm not sure *how* it works, but before I added this there were some weird errors happening on __del__. Maybe `hasattr` returns false if the property isn't initialized or something.

Reply:
Update: as we found later, it definitely doesn't work. I pushed a couple more commits that fix it, but I think I'll have to open a separate PR for that. Sorry!